lib/child_process.js

```diff
@@ -245,39 +243,29 @@ exports.execFile = function(file /*, args, options, callback*/) {
   }
 
   if (child.stdout) {
-    if (encoding)
-      child.stdout.setEncoding(encoding);
-
     child.stdout.addListener('data', function(chunk) {
-      stdoutLen += chunk.length;
+      stdoutLen += (chunk.byteLength || chunk.length);
```

Comment:
As I previously mentioned, you should be able to use `Buffer.byteLength()` for strings. AFAIK this would allow you to even keep the original `setEncoding()` call. You can also see an alternative solution [here](https://github.com/nodejs/node/pull/6764#discussion_r63407418).